libcudacxx/include/cuda/std/__random/weibull_distribution.h

```diff
@@ -0,0 +1,189 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------====//
+
+#ifndef _CUDA_STD___WEIBULL_DISTRIBUTION_H
+#define _CUDA_STD___WEIBULL_DISTRIBUTION_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/__cmath/exponential_functions.h>
+#include <cuda/std/__limits/numeric_limits.h>
+#include <cuda/std/__random/exponential_distribution.h>
+#include <cuda/std/__random/is_valid.h>
+
+#if !_CCCL_COMPILER(NVRTC)
+#  include <iosfwd>
+#endif // !_CCCL_COMPILER(NVRTC)
+
+#include <cuda/std/__cccl/prologue.h>
+
+_CCCL_BEGIN_NAMESPACE_CUDA_STD
+
+template <class _RealType = double>
+class weibull_distribution
+{
+  static_assert(__libcpp_random_is_valid_realtype<_RealType>, "RealType must be a supported floating-point type");
+
+public:
+  // types
+  using result_type = _RealType;
+
+  class param_type
+  {
+  private:
+    result_type __a_ = result_type{1};
+    result_type __b_ = result_type{1};
+
+  public:
+    using distribution_type = weibull_distribution;
+
+    _CCCL_API constexpr explicit param_type(result_type __a = result_type{1}, result_type __b = result_type{1}) noexcept
+        : __a_{__a}
+        , __b_{__b}
+    {}
+
+    [[nodiscard]] _CCCL_API constexpr result_type a() const noexcept
+    {
+      return __a_;
+    }
+    [[nodiscard]] _CCCL_API constexpr result_type b() const noexcept
+    {
+      return __b_;
+    }
+
+    [[nodiscard]] _CCCL_API friend constexpr bool operator==(const param_type& __x, const param_type& __y) noexcept
+    {
+      return __x.__a_ == __y.__a_ && __x.__b_ == __y.__b_;
+    }
+#if _CCCL_STD_VER <= 2017
+    [[nodiscard]] _CCCL_API friend constexpr bool operator!=(const param_type& __x, const param_type& __y) noexcept
+    {
+      return !(__x == __y);
+    }
+#endif // _CCCL_STD_VER <= 2017
+  };
+
+private:
+  param_type __p_;
```

Comment:
Q: Should this be value initialized?
```suggestion
  param_type __p_{};
```